exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -277,8 +279,23 @@ def _translate_attributes(self, attributes) -> TypingSequence[KeyValue]:
                     logger.exception(error)
         return output
 
-    def _export(self, data: TypingSequence[SDKDataT]) -> ExportResultT:
-
+    def _export(
+        self, data: Union[TypingSequence[ReadableSpan], MetricsData]
+    ) -> ExportResultT:
+
+        # FIXME remove this check if the export type for traces
+        # gets updated to a class that represents the proto
+        # TracesData and use the code below instead.
+        # logger.warning(
+        #     "Transient error %s encountered while exporting %s, retrying in %ss.",
+        #     error.code(),
+        #     data.__class__.__name__,
+        #     delay,
+        # )
+        if isinstance(data, TypingSequence):
+            exporting = "traces"
+        else:
+            exporting = "metrics"
```

Comment:
There is also a logs exporter

Reply:
Considering that logs is still underscore-prefixed, would this be a good opportunity to refactor the exported object to match the OTLP log proto as we did for metrics?

Reply:
Yeah, we can do that.

Reply:
Could be in a different PR IMO so non-blocking.

Reply:
There are log exporters for both grpc and http. The if block needs to be updated to check if the first element of sequence is `ReadableSpan` or `LogData` otherwise the message is incorrect.

Reply:
@srikanthccv opened #2800 to track the refactor of the exported log message.

Reply:
@srikanthccv refactored this to use a specific subclass property. This should be better because it removes the hardcoding of strings in the parent class.